code/string_algorithms/src/reverse_word_string/reverse_word_string.cpp

```diff
@@ -0,0 +1,48 @@
+// Reverse the words in a given string
+#include <algorithm>
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+#include <vector>
+
+void reverseWords(std::string s) {
+    std::string temp;
+    int c = 1;
+    s.resize(s.length());
```

Comment:
What is the purpose of this line?

Reply:
I was able to assign s to temp index by index only after resizing the strings

Reply:
@HariniJeyaraman I've tried out your code online and it [works fine](https://repl.it/repls/IllegalMoccasinAbstracttype) with this line commented out. Also, I would like that you remove the extra spaces between each of the words when printing. Some words have 2, others have 1 space in between.